File: tools/uuid_checker.py
#!/usr/bin/python3

# Python version of uuid_checker.go

import uuid
import time

start_time = time.time()

collisions = {}
collisionCount = 0
summaryEvery = 1000000
summaryCounter = 0
errors = 0
count = 0

while True:
    u = uuid.uuid4()

    if u in collisions:
        uCount = collisions[u] + 1
        collisions[u] = uCount
        print(f"{uCount} collisions for UUID {u}")
    else:
        collisions[u] = 0

    count += 1
    summaryCounter += 1
    if summaryCounter >= summaryEvery:
        summaryCounter = 0
        runningTime = time.time() - start_time
        print(f"Summary: count={count}, errors={errors}, collisions={collisionCount}, runningTime={runningTime}s")

File: tools/uuid_checker.go
package main

import (
	"fmt"
	"github.com/google/uuid"
)

func main() {

	collisions := make(map[uuid.UUID]int)
	collisionCount := 0
	summaryEvery := 10000000
	summaryCounter := 0
	errors := 0
	var count int64

	for {
		u, err := uuid.NewRandom()

		if err == nil {
			if count, ok := collisions[u]; ok {
				collisions[u] = count + 1
				collisionCount++
				fmt.Printf("%d collisions for UUID %v\n", collisions[u], u)
			} else {
				collisions[u] = 0
			}
		} else {
			fmt.Println("Error: %v\n", err)
		}

		count++
		summaryCounter++
		if summaryCounter >= summaryEvery {
			summaryCounter = 0
			fmt.Printf("Summary: count=%v, errors=%v, collisions=%v\n", count, errors, collisionCount)
		}
	}

}
